python/lsst/pex/config/config.py

```diff
@@ -1382,6 +1404,43 @@ def compare(self, other, shortcut=True, rtol=1E-8, atol=1E-8, output=None):
         return compareConfigs(name, self, other, shortcut=shortcut,
                               rtol=rtol, atol=atol, output=output)
 
+    @classmethod
+    def __init_subclass__(cls, **kwargs):
+        """Run initialization for every subclass.
+
+        Specifically registers the subclass with a YAML representer
+        and YAML constructor (if pyyaml is available)
+        """
+        super().__init_subclass__(**kwargs)
+
+        if not yaml:
+            return
+
+        def config_representer(dumper, data):
+            """Represent a Config object in a form suitable for YAML.
+
+            Stores the serialized stream as a sequence of strings.
+            """
+            stream = io.StringIO()
+            data.saveToStream(stream)
+            configLines = stream.getvalue().split("\n")
```

Comment:
If you could use the "pipe-style" multiline string representation, that would be much nicer, but I don't know if you can control that from within a representer.

Reply:
I have enabled pipe mode. It was not as straight forward as it should have been because you can only specify the text mode to use in represent_scalar. Sine we need to use represent_mapping to store the class name, I had to reimplement represent_mapping and assign an explicit ScalarNode with pipe enabled. It does work.   It really makes me want to look at ruaml...